probdist/binomial.go

```diff
@@ -1,28 +1,39 @@
 package stats
 
 import (
+	"errors"
 	"math"
 )
 
-// Binomial is used to represent the binomial distribution parameters
-// N is the number of trials {0, 1, 2, 3, ...}
-// P is the succes probability of each trial [0, 1]
+// Binomial is used to represent the binomial distribution parameters.
+// N is the number of trials {0, 1, 2, 3, ...}.
+// P is the succes probability of each trial [0, 1].
 type Binomial struct {
 	N float64
 	P float64
 }
 
-// Mean returns the mean of the binomial distribution
+// NewBinomial is used to initialize binomial parameters. What is different from
+// Binomial type is that here the parameters are validated.
+// N must be an integer positive number and P a real between 0 and 1.
+func NewBinomial(n float64, p float64) (Binomial, error) {
+	if math.Mod(n, 1.0) != 0 || p < 0 || p > 1.0 {
+		return Binomial{20.0, 0.5}, errors.New("stats: invalid Binomial parameters")
```

Comment:
Why do you initialize the struct? Just return Binomial{}.

Same applies to other constructors.